hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/TestPrefetch.java

```diff
@@ -298,6 +299,44 @@ public void testPrefetchDoesntSkipRefs() throws Exception {
     });
   }
 
+  @Test
+  public void testOnConfigurationChange() {
+    // change PREFETCH_DELAY_ENABLE_KEY from false to true
+    conf.setInt(PREFETCH_DELAY, 40000);
+    PrefetchExecutor.loadConfiguration(conf);
+    assertTrue(getPrefetchDelay() == 40000);
+
+    // restore
+    conf.setInt(PREFETCH_DELAY, 30000);
+    PrefetchExecutor.loadConfiguration(conf);
+    assertTrue(getPrefetchDelay() == 30000);
+
+  }
+
+  @Test
+  public void testPrefetchWithDelay() throws Exception {
```

Comment:
Can we assert that 1) at first, no Prefetch thread is running, 2) After the configured delay has elapsed, prefetch thread runs?

Reply:
Captured the timing after 
    threads are scheduled with the delay and
    before threads start executing 
This will ensure that Prefetch waited for the specified delay

Reply:
You know your delay, and you know that no threads should be running before this delay. You can easily get the scheduled threads from PrefetchExecutor via prefetchFutures/prefetchRunnables and use it to test the state of the thread prior and after the delay has elapsed. That way you can test this without the need to expose a non-thread safe long with the "computed" delay for test purposes only (an anti-pattern).